SingularityUI/app/components/common/table/UITable.jsx

```diff
@@ -12,19 +11,35 @@ class UITable extends Component {
   constructor(props) {
     super(props);
 
+    let { data } = props;
+    const { defaultSortBy, defaultSortDirection, rowChunkSize } = props;
+    if (defaultSortBy) {
+      data = this.doSort(data, defaultSortBy, defaultSortDirection);
+    }
+
     this.state = {
-      sortBy: props.defaultSortBy,
-      sortDirection: props.defaultSortDirection,
+      sortBy: defaultSortBy,
+      sortDirection: defaultSortDirection,
       sortTime: null,
       chunkNum: 1,
-      data: props.data
+      data,
+      rowChunkSize
     };
 
     this.handlePageChange = this.handlePageChange.bind(this);
   }
 
   componentWillReceiveProps(nextProps) {
     this.updateSort(nextProps.data, this.state.sortBy, this.state.sortDirection);
+    if (nextProps.isFetching) {
+      return;
+    }
+    if (this.isServerSideTable() && _.isEmpty(nextProps.data) && this.state.chunkNum > 1) {
+      this.fetchDataFromApi(this.state.chunkNum - 1, this.state.rowChunkSize, this.state.sortBy);
+      this.setState({ pastEnd: true });
+    } else if (this.isServerSideTable() && (this.state.pastEnd || nextProps.data.length < this.state.rowChunkSize)) {
+      this.setState({pastEnd: false, lastPage: true});
```

Comment:
I know we don't lint spacing on object literals, but I think we should maintain consistency here